src/java.base/share/classes/java/lang/Float16.java

```diff
@@ -638,6 +638,7 @@ public short shortValue() {
      * @jls 5.1.3 Narrowing Primitive Conversion
      */
     @Override
+    @IntrinsicCandidate
     public int intValue() {
         return (int)floatValue();
```

Comment:
Should it be handled trough Idealization xform ?

ConvHF2F + ConvF2I => ConvHF2I

Reply:
Matcher pattern may also suffice, but the problem will if ConvHF2F has multiple users.

Reply:
If you check JDK mainline Float16 C2 compiler support PR, John has suggested us to go with pattern matching, I have added more details to that draft PR.

Reply:
Hi Jatin, thanks for the review comments. I did go through the comments on your draft PR earlier but I wasn't sure if we would be following their comments on the Valhalla branch as yet but I do agree with him about having too many intrinsics being an overkill. I'll remove the `@IntrinsicCandidate` here and add the code to do pattern matching in the mid-end.